Update docs to replace `createInteractor` in favor of `extends` (#921)
thefrontside/bigtest: website/docs/interactors/6-write-your-own.md

3. The locator and filters, which helps users be able to narrow down the element they want to reference
4. The action or actions that a test should `perform` when using the interactor (like `click`)

> Putting this together, let's create a new Interactor called 'MyTextField' with a label of 'my-textfield-interactor'. We'll specify the selector as `input[type=text]` so that it targets all the text input elements, define a `value` filter, and provide a `fillIn` action. And to differentiate from the built-in TextField interactor, we'll configure the placeholder value as its locator:

Author: This paragraph wasn't lining up with the example so I re-wrote it. `In this case` didn't really make any sense because the bullet points above it applies to _all_ cases. I think the rewrite is more concise.

:::note
`fillIn` is a function exported by `bigtest`. See the implementation [here](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/fill-in.ts#L63-L76). You can use any of the functions defined by BigTest or implement your own.
:::

:::note Cypress
If you're using Cypress, all of the built-in Interactors and Interactor functions will need to be imported from `@bigtest/cypress` and not `bigtest`.
:::

### `extend()` method

In the example above, we're extending from the `HTML` interactor to compose the `MyTextField` interactor, but if you take a look at the implementation of the [built-in TextField](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/definitions/text-field.ts) interactor, you'll see that the `value` filter and `fillIn` action are already available.

> You could reimplement the `value` filter and `fillIn` action from scratch like we did in the example, however, it would be more practical to just extend from the built-in TextField interactor instead:

Author: Hmm...

</Tabs>

:::note
`fillIn` is a function exported by `bigtest`. See the implementation [here](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/fill-in.ts#L63-L76). You can use any of the functions defined by BigTest or implement your own.
:::

> In the example above we're extending from the `HTML` interactor to compose the MyTextField interactor. If you take a look at the implementation of the built-in [TextField](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/definitions/text-field.ts) interactor, you'll see that the `value` filter and `fillIn` action is already available.

Reviewer: ```suggestion
In the example above, we're extending from the `HTML` interactor to compose the MyTextField interactor. If you take a look at the implementation of the built-in [TextField](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/definitions/text-field.ts) interactor, you'll see that the `value` filter and `fillIn` action are already available.
```

> In a more realistic scenario, if you want a TextField interactor that uses its placeholder value as its locator, you would just extend from the pre-existing TextField interactor like so:

Reviewer: ```suggestion
In a more realistic scenario, if you want a TextField interactor that uses its placeholder value as its locator, you should extend from the pre-existing TextField interactor:
```

```js
import { TextField } from 'bigtest';

export const MyTextField = TextField.extend('my-textfield-interactor')
  .locator((element) => element.placeholder)
```

> This approach would allow your interactor to inherit the selector, locator, filters, and actions from the interactor it is extending from. Extending from another interactor provides the convenience of not having to rewrite common filters and actions. You can overwrite any of the inherited properties to suit your needs which is what we are doing with the locator in the example.

Reviewer: ```suggestion
This approach would allow your interactor to inherit the selector, locator, filters, and actions from the interactor it is extending from. Extending from another interactor provides the convenience of not having to rewrite common filters and actions. You can overwrite any of the inherited properties to suit your needs, which is what we are doing with the locator in the example.
```

If we didn't provide the placeholder value as the locator, it would inherit the locator from BigTest's built-in TextField interactor which itself is inheriting its locator from the [FormField](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/definitions/form-field.ts#L6) interactor.

> 

Author: I'm not sure if this is an appropriate spot to discuss what I wrote here. Please take a look at the generated Netlify preview [here](https://deploy-preview-921--bigtest.netlify.app/bigtest/docs/interactors/write-your-own/) to see if it flows naturally.

Reviewer 2: I think using 'inherit' is getting too into the technical detail. I would just say 'use' instead of 'inherit' in casual conversation. I don't think we should be drilling inheritance into people, it's just one of the many mechanisms behind the APIs.

Author: Yeah, reading it through again now, I say `inherit` way too much. I'll rewrite it once we decide on what to do with `HTML` on the built-in page.

Author: See new update down below.

  .locator((element) => element.placeholder)
```

> This approach would allow `MyTextField` to inherit the selector, locator, filters, and actions from the built-in `TextField`. You can overwrite any of the inherited properties to suit your needs, which is what we are doing with the locator in this instance.

Author: I think this rewrite makes the point much more concise?

Author: This one isn't outdated either.

  ```js
  import { Button, Heading, Page, test } from 'bigtest';
  import { MyTextField } from './MyTextField';

  export default test('email subscription form')
    .step(Page.visit('/'))
    .child('fill and submit email address', test => test
      .step(
        MyTextField('EMAIL').fillIn('batman@gmail.com'),
        Button('Subscribe').click())
      .assertion(Heading('Success!').exists()));
  ```

  </TabItem>
</Tabs>

By composing our own text field interactor, we are able to use it with its new locator value without which we would have needed to use a filter for each instance:

```js
TextField({ placeholder: 'email' }).fillIn('batman@gmail.com');
MyTextField('email').fillIn('batman@gmail.com');
```

> Although the built-in interactors may cover most use cases, the composability of interactors means there is no limit to how much you can optimize and tailor them to your needs.

Author: New approach to wrapping up the first example.

This example uses [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) which is available in Node >=14.
:::

> Once again, by extending from the `HTML` interactor, our new TableCell interactor inherits all of the pre-defined filters and actions of the `HTML` interactor as defined [here](https://github.com/thefrontside/bigtest/blob/v0/packages/interactor/src/definitions/html.ts). If we needed to access a table cell's `id` property in our tests, we would not need to create a separate filter for it as it inherits it from the `HTML` interactor.

Author: I don't know if this is needs to be reiterated. I might be fixating on the obvious too much.

Reviewer 2: I think it's ok
